File: src/main/java/com/example/SpringServer/repository/TypeRepository.java
package com.example.SpringServer.repository;

import com.example.SpringServer.Entities.Type;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import javax.transaction.Transactional;

@Repository
@Transactional
public interface TypeRepository extends CrudRepository<Type, Long>, TypeRepositoryCustom {}

File: src/main/kotlin/de/lrprojects/tempserver/repository/TypeRepository.kt
package de.lrprojects.tempserver.repository

import de.lrprojects.tempserver.entity.Type
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface TypeRepository : JpaRepository<Type, Long> {

    fun findByTypeName(typeName: String): Type?
}